x/collection/keeper/msg_server_test.go

```diff
@@ -268,109 +338,155 @@ func (s *KeeperTestSuite) TestMsgOperatorSendNFT() {
+			req: &collection.MsgRevokeOperator{
+				ContractId: s.contractID,
+				Holder:     s.customer.String(),
+				Operator:   s.operator.String(),
+			},
+			expectedEvent: helperBuildEventAuthRelated(revokedOperatorEventType, s.contractID, s.customer, s.operator),
 		},
 		"contract not found": {
-			contractID: "deadbeef",
-			holder:     s.customer,
-			operator:   s.operator,
-			err:        class.ErrContractNotExist,
+			isNegativeCase: true,
+			req: &collection.MsgRevokeOperator{
+				ContractId: "deadbeef",
+				Holder:     s.customer.String(),
+				Operator:   s.operator.String(),
+			},
+			expectedError: class.ErrContractNotExist,
 		},
 		"no authorization": {
-			contractID: s.contractID,
-			holder:     s.customer,
-			operator:   s.vendor,
-			err:        collection.ErrCollectionNotApproved,
+			isNegativeCase: true,
+			req: &collection.MsgRevokeOperator{
+				ContractId: s.contractID,
+				Holder:     s.customer.String(),
+				Operator:   s.vendor.String(),
+			},
+			expectedError: collection.ErrCollectionNotApproved,
 		},
 	}
 
 	for name, tc := range testCases {
 		s.Run(name, func() {
-			ctx, _ := s.ctx.CacheContext()
+			// Arrange
+			prevEvtCnt := len(s.ctx.EventManager().Events())
+			holder, err := sdk.AccAddressFromBech32(tc.req.Holder)
+			s.Require().NoError(err)
+			operator, err := sdk.AccAddressFromBech32(tc.req.Operator)
+			s.Require().NoError(err)
+			prevAuth, _ := s.keeper.GetAuthorization(s.ctx, tc.req.ContractId, holder, operator)
 
-			req := &collection.MsgRevokeOperator{
-				ContractId: tc.contractID,
-				Holder:     tc.holder.String(),
-				Operator:   tc.operator.String(),
-			}
-			res, err := s.msgServer.RevokeOperator(sdk.WrapSDKContext(ctx), req)
-			s.Require().ErrorIs(err, tc.err)
-			if tc.err != nil {
+			// Act
+			res, err := s.msgServer.RevokeOperator(sdk.WrapSDKContext(s.ctx), tc.req)
+			if tc.isNegativeCase {
+				s.Require().ErrorIs(err, tc.expectedError)
 				return
 			}
-
+			s.Require().NoError(err)
 			s.Require().NotNil(res)
 
-			if s.deterministic {
-				s.Require().Equal(tc.events, ctx.EventManager().Events())
-			}
+			// Assert
+			events := s.ctx.EventManager().Events()
+			lastEvent := events[len(events)-1]
+			s.Require().Equal(tc.expectedEvent, lastEvent)
+			s.Require().Greater(len(s.ctx.EventManager().Events()), prevEvtCnt)
+			s.Require().NotNil(prevAuth)
+			s.Require().Equal(tc.req.Holder, prevAuth.Holder)
+			s.Require().Equal(tc.req.Operator, prevAuth.Operator)
+			curAuth, err := s.keeper.GetAuthorization(s.ctx, tc.req.ContractId, holder, operator)
+			s.Require().ErrorIs(err, collection.ErrCollectionNotApproved)
+			s.Require().Nil(curAuth)
 		})
 	}
 }
 
+func helperBuildEventAuthRelated(evtType, contractID string, holder, operator sdk.AccAddress) sdk.Event {
```

Comment:
To prevent event-breaking, I should add unit test for `helperBuildEventAuthRelated` with the original literal events to verify event-breaking. 
Do I understand the point? If so, I'll add unit test for it.

Reply:
> I should add unit test for `helperBuildEventAuthRelated` with the original literal events to verify event-breaking.

:+1: And also for other helpers.

I forgot to mention about the other events. It seems like the new tests check only the last event. So you need to add a line to check the last event is the only event caused by the request, hence there is no other event.